# Initial census tract data in geopandas

In [1]:
import geoplot as gplt
import geopandas as gpd
import geoplot.crs as gcrs
import imageio
import folium
import requests
import pathlib

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import mapclassify as mc

%matplotlib inline
pd.options.display.max_rows = 999

In [2]:
census_tract = gpd.read_file('./data/census_tract_2010_cook_county/tl_2010_17031_tract10.shp')
census_tract.head()

STATEFP10 COUNTYFP10 TRACTCE10      GEOID10 NAME10         NAMELSAD10  \
0        17        031    840300  17031840300   8403  Census Tract 8403   
1        17        031    840200  17031840200   8402  Census Tract 8402   
2        17        031    841100  17031841100   8411  Census Tract 8411   
3        17        031    841200  17031841200   8412  Census Tract 8412   
4        17        031    838200  17031838200   8382  Census Tract 8382   

  MTFCC10 FUNCSTAT10  ALAND10  AWATER10   INTPTLAT10    INTPTLON10  \
0   G5020          S   831224         0  +41.8320943  -087.6818822   
1   G5020          S   831087         0  +41.8445748  -087.6491915   
2   G5020          S  1143247         0  +41.8510058  -087.6350978   
3   G5020          S   625405         0  +41.8555618  -087.6833420   
4   G5020          S  1157564         0  +41.8704157  -087.6750794   

                                            geometry  
0  POLYGON ((-87.68608 41.82296, -87.68607 41.823...  
1  POLYGON ((-87.64633 41.84009, -87.64632 41.839...  
2  POLYGON ((-87.62935 41.85280, -87.62934 41.852...  
3  POLYGON ((-87.68814 41.85569, -87.68816 41.856...  
4  POLYGON ((-87.66782 41.87418, -87.66768 41.874...

In [3]:
census_tract.shape

NameError: name 'census_track' is not defined

In [4]:
type(census_tract['GEOID10'][0])

str

In [5]:
census_tract['GEOID10'].isna().sum()

0

The original food access data is too large to host on the GitHub repo. Click here to access [the Food Access Research Atlas Data Download 2019](https://www.ers.usda.gov/data-products/food-access-research-atlas/download-the-data/). 

In [6]:
food_access = pd.read_csv('./food_access/FoodAccessResearchAtlasData2019.csv')
food_access.head()

CensusTract    State          County  Urban  Pop2010  OHU2010  \
0   1001020100  Alabama  Autauga County      1     1912      693   
1   1001020200  Alabama  Autauga County      1     2170      743   
2   1001020300  Alabama  Autauga County      1     3373     1256   
3   1001020400  Alabama  Autauga County      1     4386     1722   
4   1001020500  Alabama  Autauga County      1    10766     4082   

   GroupQuartersFlag  NUMGQTRS  PCTGQTRS  LILATracts_1And10  ...  \
0                  0       0.0      0.00                  0  ...   
1                  0     181.0      8.34                  1  ...   
2                  0       0.0      0.00                  0  ...   
3                  0       0.0      0.00                  0  ...   
4                  0     181.0      1.68                  0  ...   

   TractSeniors  TractWhite  TractBlack  TractAsian  TractNHOPI  TractAIAN  \
0         221.0      1622.0       217.0        14.0         0.0       14.0   
1         214.0       888.0      1217.0         5.0         0.0        5.0   
2         439.0      2576.0       647.0        17.0         5.0       11.0   
3         904.0      4086.0       193.0        18.0         4.0       11.0   
4        1126.0      8666.0      1437.0       296.0         9.0       48.0   

   TractOMultir  TractHispanic  TractHUNV  TractSNAP  
0          45.0           44.0        6.0      102.0  
1          55.0           75.0       89.0      156.0  
2         117.0           87.0       99.0      172.0  
3          74.0           85.0       21.0       98.0  
4         310.0          355.0      230.0      339.0  

[5 rows x 147 columns]

In [7]:
type(food_access['CensusTract'][0])

numpy.int64

In [8]:
food_access['CensusTract'].isna().sum()

0

In [9]:
food_access['CensusTract'][0]

1001020100

In [10]:
# convert food_access['CensusTract'] to a string
food_access['CensusTract']= [str(ct_num) for ct_num in food_access['CensusTract']]

In [11]:
type(food_access['CensusTract'][0])

str

In [12]:
food_access['CensusTract'][0]

'1001020100'

In [13]:
food_access.head()

CensusTract    State          County  Urban  Pop2010  OHU2010  \
0  1001020100  Alabama  Autauga County      1     1912      693   
1  1001020200  Alabama  Autauga County      1     2170      743   
2  1001020300  Alabama  Autauga County      1     3373     1256   
3  1001020400  Alabama  Autauga County      1     4386     1722   
4  1001020500  Alabama  Autauga County      1    10766     4082   

   GroupQuartersFlag  NUMGQTRS  PCTGQTRS  LILATracts_1And10  ...  \
0                  0       0.0      0.00                  0  ...   
1                  0     181.0      8.34                  1  ...   
2                  0       0.0      0.00                  0  ...   
3                  0       0.0      0.00                  0  ...   
4                  0     181.0      1.68                  0  ...   

   TractSeniors  TractWhite  TractBlack  TractAsian  TractNHOPI  TractAIAN  \
0         221.0      1622.0       217.0        14.0         0.0       14.0   
1         214.0       888.0      1217.0         5.0         0.0        5.0   
2         439.0      2576.0       647.0        17.0         5.0       11.0   
3         904.0      4086.0       193.0        18.0         4.0       11.0   
4        1126.0      8666.0      1437.0       296.0         9.0       48.0   

   TractOMultir  TractHispanic  TractHUNV  TractSNAP  
0          45.0           44.0        6.0      102.0  
1          55.0           75.0       89.0      156.0  
2         117.0           87.0       99.0      172.0  
3          74.0           85.0       21.0       98.0  
4         310.0          355.0      230.0      339.0  

[5 rows x 147 columns]

In [14]:
food_access[food_access['County'] == 'Cook County'].shape

(1320, 147)

### Merge census_tract and food_access on CensusTract and GEOID10 including only the data that exist in both files.

In [16]:
food_data = food_access.merge(census_tract, left_on = 'CensusTract', right_on = 'GEOID10', how = 'inner')
food_data.head()

CensusTract     State       County  Urban  Pop2010  OHU2010  \
0  17031010100  Illinois  Cook County      1     4854     2302   
1  17031010201  Illinois  Cook County      1     6450     2463   
2  17031010202  Illinois  Cook County      1     2818     1115   
3  17031010300  Illinois  Cook County      1     6236     2826   
4  17031010400  Illinois  Cook County      1     5042     2098   

   GroupQuartersFlag  NUMGQTRS  PCTGQTRS  LILATracts_1And10  ...      GEOID10  \
0                  0     218.0      4.49                  0  ...  17031010100   
1                  0     163.0      2.53                  0  ...  17031010201   
2                  0     315.0     11.18                  0  ...  17031010202   
3                  0     791.0     12.68                  0  ...  17031010300   
4                  0    1349.0     26.76                  0  ...  17031010400   

   NAME10           NAMELSAD10  MTFCC10  FUNCSTAT10  ALAND10  AWATER10  \
0     101     Census Tract 101    G5020           S   381061         0   
1  102.01  Census Tract 102.01    G5020           S   504470         0   
2  102.02  Census Tract 102.02    G5020           S   351716         0   
3     103     Census Tract 103    G5020           S   472017         0   
4     104     Census Tract 104    G5020           S   548754    602605   

    INTPTLAT10    INTPTLON10  \
0  +42.0212617  -087.6698442   
1  +42.0160076  -087.6801485   
2  +42.0160503  -087.6733216   
3  +42.0159407  -087.6665351   
4  +42.0064028  -087.6571350   

                                            geometry  
0  POLYGON ((-87.66368 42.01940, -87.66384 42.019...  
1  POLYGON ((-87.68009 42.01254, -87.68027 42.012...  
2  POLYGON ((-87.67336 42.01937, -87.67311 42.019...  
3  POLYGON ((-87.66506 42.01280, -87.66543 42.012...  
4  POLYGON ((-87.65080 41.99848, -87.65500 41.998...  

[5 rows x 160 columns]

In [17]:
food_data.shape

(1314, 160)

#### census tract has 1319 rows, food_access has 1320 rows. When merged food_data has 1314 rows.

In [18]:
food_data.isna().sum().sort_values(ascending = False)

lasnap20share           1314
laasian10               1314
lasnap10                1314
lahunv10share           1314
lahisp10share           1314
lahisp10                1314
laomultir10share        1314
laomultir10             1314
laaian10share           1314
laaian10                1314
lanhopi10share          1314
lanhopi10               1314
laasian10share          1314
lablack10share          1314
lapop20                 1314
lablack10               1314
lawhite10share          1314
lawhite10               1314
laseniors10share        1314
laseniors10             1314
lakids10share           1314
lakids10                1314
lalowi10share           1314
lalowi10                1314
lapop10share            1314
lapop10                 1314
lasnap10share           1314
lahunv10                1314
laaian20                1314
lahisp20                1314
laomultir20             1314
laaian20share           1314
lahisp20share           1314
lanhopi20share          1314
lanhopi20     

In [19]:
food_data.describe().T

count          mean           std      min  \
Urban                 1314.0  9.984779e-01  3.899886e-02      0.0   
Pop2010               1314.0  3.953330e+03  1.788800e+03    237.0   
OHU2010               1314.0  1.496466e+03  7.610063e+02    100.0   
GroupQuartersFlag     1314.0  3.805175e-03  6.159207e-02      0.0   
NUMGQTRS              1314.0  6.870776e+01  3.525771e+02      0.0   
PCTGQTRS              1314.0  1.738508e+00  6.321919e+00      0.0   
LILATracts_1And10     1314.0  3.881279e-02  1.932221e-01      0.0   
LILATracts_halfAnd10  1314.0  2.519026e-01  4.342709e-01      0.0   
LILATracts_1And20     1314.0  3.881279e-02  1.932221e-01      0.0   
LILATracts_Vehicle    1314.0  1.415525e-01  3.487233e-01      0.0   
HUNVFlag              1314.0  2.009132e-01  4.008359e-01      0.0   
LowIncomeTracts       1314.0  5.182648e-01  4.998565e-01      0.0   
PovertyRate           1314.0  1.714072e+01  1.293260e+01      0.3   
MedianFamilyIncome    1305.0  8.304176e+04  4.820180e+04  10131.0   
LA1and10              1314.0  1.278539e-01  3.340542e-01      0.0   
LAhalfand10           1314.0  5.684932e-01  4.954750e-01      0.0   
LA1and20              1314.0  1.278539e-01  3.340542e-01      0.0   
LATracts_half         1314.0  5.684932e-01  4.954750e-01      0.0   
LATracts1             1314.0  1.278539e-01  3.340542e-01      0.0   
LATracts10            1314.0  0.000000e+00  0.000000e+00      0.0   
LATracts20            1314.0  0.000000e+00  0.000000e+00      0.0   
LATractsVehicle_20    1314.0  2.009132e-01  4.008359e-01      0.0   
LAPOP1_10              340.0  1.124400e+03  1.561402e+03      0.0   
LAPOP05_10             933.0  2.167891e+03  1.790938e+03      0.0   
LAPOP1_20              340.0  1.124400e+03  1.561402e+03      0.0   
LALOWI1_10             340.0  2.699206e+02  4.355477e+02      0.0   
LALOWI05_10            933.0  6.421726e+02  6.526412e+02      0.0   
LALOWI1_20             340.0  2.699206e+02  4.355477e+02      0.0   
lapophalf              935.0  2.173412e+03  1.793004e+03      0.0   
lapophalfshare         935.0  5.221407e+01  3.429060e+01      0.0   
lalowihalf             935.0  6.440620e+02  6.548516e+02      0.0   
lalowihalfshare        935.0  1.727427e+01  1.786605e+01      0.0   
lakidshalf             935.0  5.358064e+02  4.612156e+02      0.0   
lakidshalfshare        935.0  1.299966e+01  9.366921e+00      0.0   
laseniorshalf          935.0  2.942246e+02  2.822096e+02      0.0   
laseniorshalfshare     935.0  7.046963e+00  5.896531e+00      0.0   
lawhitehalf            935.0  1.249034e+03  1.453662e+03      0.0   
lawhitehalfshare       935.0  2.688809e+01  2.731286e+01      0.0   
lablackhalf            935.0  6.002000e+02  1.023286e+03      0.0   
lablackhalfshare       935.0  1.814759e+01  2.957736e+01      0.0   
laasianhalf            935.0  1.301807e+02  2.888110e+02      0.0   
laasianhalfshare       935.0  2.702428e+00  5.096662e+00      0.0   
lanhopihalf            935.0  5.700535e-01  1.464484e+00      0.0   
lanhopihalfshare       935.0  1.269519e-02  3.265890e-02      0.0   
laaianhalf             935.0  6.812834e+00  1.063338e+01      0.0   
laaianhalfshare        935.0  1.635615e-01  2.419807e-01      0.0   
laomultirhalf          935.0  1.865882e+02  3.062937e+02      0.0   
laomultirhalfshare     935.0  4.299690e+00  6.434302e+00      0.0   
lahisphalf             935.0  3.429743e+02  6.143459e+02      0.0   
lahisphalfshare        935.0  7.786824e+00  1.265474e+01      0.0   
lahunvhalf             935.0  8.903316e+01  1.260586e+02      0.0   
lahunvhalfshare        935.0  7.380888e+00  1.098839e+01      0.0   
lasnaphalf             935.0  1.152150e+02  1.425276e+02      0.0   
lasnaphalfshare        935.0  9.325112e+00  1.258870e+01      0.0   
lapop1                 342.0  1.133763e+03  1.562099e+03      0.0   
lapop1share            342.0  2.305336e+01  2.745036e+01      0.0   
lalowi1                342.0  2.727047e+02  4.372589e+02      0.0   
lalowi1share         

In [20]:
# drop the geometry column for now and then write data to csv
food_data.drop(columns = 'geometry').to_csv('./data/food_data.csv')